### PR TITLE
test: make all tests run on windows

### PR DESCRIPTION
**Pull request by tommywo**

### CODE REVIEW

**Comment by stevemao** [comment]
Great work! Thanks for the effort!

File: .travis.yml [hunk 1]
```diff
@@ -15,14 +18,15 @@ cache:
     - ${HOME}/.npm
 
 before_install:
-  - npm install --global npm@latest
```
Comment (stevemao) [inline .travis.yml]: I'm wondering if it's necessary to do this on Mac or Linux? If not, we can remove the whole thing.

File: package.json [hunk 2]
```diff
@@ -95,5 +98,16 @@
...
     "standard-changelog": "file:packages/standard-changelog"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "*.js": [
+      "eslint --fix --cache --cache-location ./node_modules/.cache/ --ignore-path .gitignore",
```
Comment (stevemao) [inline package.json]: TIL: `--ignore-path .gitignore` nice trick :)

File: .travis.yml (hunk 1 above)
Comment (tommywo) [reply .travis.yml]: I think that was included to use `npm ci`, so if we test only node version that have npm >=5.7.0 it should be safe to remove.

File: packages/conventional-changelog-jquery/test/test.js [hunk 3]
```diff
@@ -19,19 +19,12 @@ describe('jquery preset', function () {
...
-    shell.exec('git add --all && git commit -m"Event: Remove an internal argument to the on method" -m"Fixes #2, #4, gh-200"')
-    writeFileSync('test4', '')
-    shell.exec('git add --all && git commit -m"Manipulation: Remove an internal argument to the remove method" -m"Closes #22"')
-    writeFileSync('test5', '')
-    shell.exec('git add --all && git commit -m"Bad commit"')
-    writeFileSync('test6', '')
-    shell.exec('git add --all && git commit -m"Core: Create jQuery.ajax" -m"Closes gh-100"')
+    gitDummyCommit(['Core: Make jQuery objects iterable'])
+    gitDummyCommit(['CSS: Don\'t name the anonymous swap function'])
+    gitDummyCommit(['Event: Remove an internal argument to the on method" -m"Fixes #2, #4, gh-200'])
```
Comment (tommywo) [inline packages/conventional-changelog-jquery/test/test.js]:
```suggestion
    gitDummyCommit(['Event: Remove an internal argument to the on method', 'Fixes #2, #4, gh-200'])
```

File: packages/conventional-changelog-jquery/test/test.js [hunk 4]
```diff
@@ -19,19 +19,12 @@ describe('jquery preset', function () {
...
+    gitDummyCommit(['Manipulation: Remove an internal argument to the remove method" -m"Closes #22'])
```
Comment (tommywo) [inline packages/conventional-changelog-jquery/test/test.js]:
```suggestion
    gitDummyCommit(['Manipulation: Remove an internal argument to the remove method', 'Closes #22'])
```

File: packages/conventional-changelog-jquery/test/test.js [hunk 5]
```diff
@@ -19,19 +19,12 @@ describe('jquery preset', function () {
...
+    gitDummyCommit(['Bad commit'])
+    gitDummyCommit(['Core: Create jQuery.ajax" -m"Closes gh-100'])
```
Comment (tommywo) [inline packages/conventional-changelog-jquery/test/test.js]:
```suggestion
    gitDummyCommit(['Core: Create jQuery.ajax', 'Closes gh-100'])
```

File: .travis.yml (hunk 1 above)
Comment (stevemao) [reply .travis.yml]: Let's remove it then :)

**Pull request merged**